plugins/MauticFocusBundle/Controller/FocusController.php

```diff
@@ -2,21 +2,41 @@
 
 namespace MauticPlugin\MauticFocusBundle\Controller;
 
+use Doctrine\Persistence\ManagerRegistry;
+use Mautic\CacheBundle\Cache\CacheProvider;
 use Mautic\CoreBundle\Controller\AbstractStandardFormController;
+use Mautic\CoreBundle\Factory\MauticFactory;
+use Mautic\CoreBundle\Factory\ModelFactory;
 use Mautic\CoreBundle\Form\Type\DateRangeType;
+use Mautic\CoreBundle\Helper\CoreParametersHelper;
+use Mautic\CoreBundle\Helper\UserHelper;
+use Mautic\CoreBundle\Security\Permissions\CorePermissions;
+use Mautic\CoreBundle\Service\FlashBag;
+use Mautic\CoreBundle\Translation\Translator;
+use Mautic\FormBundle\Helper\FormFieldHelper;
 use Mautic\PageBundle\Model\TrackableModel;
 use MauticPlugin\MauticFocusBundle\Entity\Focus;
 use MauticPlugin\MauticFocusBundle\Model\FocusModel;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\Form\FormFactoryInterface;
 use Symfony\Component\HttpFoundation\JsonResponse;
 use Symfony\Component\HttpFoundation\RedirectResponse;
 use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
 use Symfony\Component\HttpFoundation\Response;
 
-/**
- * Class FocusController.
- */
 class FocusController extends AbstractStandardFormController
 {
+    private CacheProvider $cacheProvider;
+
+    /** @phpstan-ignore-next-line */
```

Comment:
What was the problem here?

Reply:
Parent class requires `MauticFactory`, which is deprecated (I'm not using it). I need to override the constructor to inject the `CacheProvider`. Do you know better workaround than just ignore this constructor?

Reply:
Ah, the deprecation warning. I was asking the same question in https://github.com/mautic/mautic/pull/12580#issuecomment-1647705911

In the future we can ignore just specific rule but it seems we'd have to upgrade PHPSTAN to support that. https://phpstan.org/user-guide/ignoring-errors So let's go with this for now.